spec/index.md

```diff
@@ -3605,6 +3630,132 @@ S with
 
 ```
 
+
+#### IC Management Canister: Upload Chunk
+
+A controller of a canister, or the canister itself can upload chunks to the chunk store of that canister.
+
+Conditions
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.method_name = 'upload_chunk'
+M.arg = candid(A)
+chunk_store_size = |{x | chunk_store[A.canister_id][x] not null}|
+chunk_store_size < CHUNK_STORE_SIZE and (M.caller ∈ S.controllers[A.canister_id] ∪ {A.canister_id})
+hash = SHA-256(A.chunk)
+M.caller ∈ S.controllers[A.canister_id] ∪ {A.canister_id}
+
+
+```
+
+State after
+
+```html
+
+S with
+    chunk_store[A.canister_id](hash') = chunk_store[A.canister_id] if hash'≠ hash
```

Comment:
I'd drop this line due to the implicit assumption that everything stays the same that we don't explicitly override in the "with" clause.

Reply:
Good point